gitbook-docs/README.md

```diff
@@ -1,31 +1,54 @@
-
-zkLLVM tool chain adds extensions via:&#x20;
-
-1. `clang` : Compiles the program into intermediate representation byte-code.
-2. `assigner` : Creates the assignment table for circuit.
-
-### Tool Chain
-
-Below we look at flow of how the zkLLVM tool chain is invoked:
-
-1. Users who wish to **generate a provable circuit**, will write their code in a compatible front end ex: C++. This code will be compiled with the modified version of the `clang` compiler , which will output byte-code representation of the circuit.  For the most performant circuit components  we recommend using crypto3 library , however this is not essential.
-2.  Users who wish to **generate a proof for the circuit** , will require to run the `assigner` along with the circuit generated above and pass the public inputs & witness (if necessary). This will output two files:
-
-    * Constraint : Binary file representing arithmetization of the circuit.
-    * Assignment Table: Binary file pre-processed with public inputs & witness.
-
-    The constraint and assignment table generated above should be passed as in input to proof generator binary. This will output a binary proof file.
-3. Proof verification is not part of the zkLLVM tool-chain currently, can be done via:&#x20;
-   1. Offline : Tooling to support validation of off-chain proof will be added in future.
-   2. On-chain : This involves a more steps which requires serialisation of the circuit and deployed on blockchain clusters. This flow of generating smart contracts is handled by the [lorem-ipsum](https://github.com/NilFoundation/lorem-ipsum-cli) project. A high level flow is described in the guides for [circuit developer](manual/getting-started/circuit-generation.md) & [proof verifier](manual/getting-started/proof-verifier.md).
+## `=nil;` zkLLVM Project
+
+zkLLVM is a compiler from high-level programming languages into input for provable computation protocols.
+Developers can use zkLLVM to generate inputs for any arbitrary zero-knowledge proof system or protocol
+that accepts input data in the form of algebraic circuits.
+It can be used with the Placeholder proof system — a derivative of PLONK —
+or any other compatible arithmetization.
+
+zkLLVM is an extension to the LLVM toolchain, supporting any frontend that compiles to
+[LLVM intermediate representation](https://llvm.org/docs/LangRef.html#introduction).
+With zkLLVM, developers can write code in native languages instead of DSLs specific to other tools.
+
+zkLLVM toolchain has two main tools:
+
+1.  `clang` compiles the program into a circuit — an intermediate representation in bytecode.
+2.  `assigner` creates an assignment table using the compiled circuit and input data.
+
+### Zero-knowledge proof workflow with zkLLVM
+
+zkLLVM toolchain is a crucial part of the zero-knowledge proof verification workflow.
+Here's how it all works.
+
+1.  First, the companies who want to prove transactions develop their code in a compatible language, like C++.
```

Comment:
`prove transactions` is incorrect and perhaps `prove computations` might be a better fit .

Reply:
Fixed.